torch/csrc/jit/codegen/cuda/runtime/helpers.cu

```diff
@@ -411,3 +411,48 @@ template <typename T>
 TypelessData<sizeof(T), alignof(T)> erase_type(T x) {
   return x;
 }
+
+template <typename T>
+bool isfinite(T x) {
+  return ::isfinite(x);
+}
+
+template <typename T>
+bool isfinite(std::complex<T> x) {
+  return ::isfinite(abs(x));
```

Comment:
This is how it is implemented in PyTorch: https://github.com/pytorch/pytorch/blob/8c47e9dc81be89b68836f6533e8035d5dee580f7/aten/src/ATen/native/TensorCompare.cpp#L223, but I don't like this implementation, because the abs of a complex could overflow. That is, both real and imag are finite, but `sqrt(real^2 + imag^2)` will be inf. For this case, we should still have `true` for `isfinite`.
cc: @mruberry

Reply:
https://github.com/pytorch/pytorch/pull/76606

Reply:
#76606 approved, let's be accurate here, too

Reply:
done